python/src/iceberg/expressions/base.py

```diff
@@ -346,3 +346,116 @@ def bind(self, schema: Schema, case_sensitive: bool) -> BoundReference:
             raise ValueError(f"Cannot find field '{self.name}' in schema: {schema}")
 
         return BoundReference(field=field, accessor=schema.accessor_for_field(field.field_id))
+
+
+class BooleanExpressionVisitor(ABC):
```

Comment:
I think this should be parameterized. The visitor will return some type, like the schema visitor.

Reply:
I see, updated to `BooleanExpressionVisitor(Generic[T], ABC)` 👍